esp-hal/src/spi/master.rs

```diff
@@ -2708,48 +2739,162 @@ impl Driver {
 
-        if ints.trans_done().bit() {
-            res.insert(SpiInterrupt::TransDone);
+                if ints.trans_done().bit() {
+                    res.insert(SpiInterrupt::TransDone);
+                }
+                if ints.dma_infifo_full_err().bit() {
+                    res.insert(SpiInterrupt::DmaInfifoFullErr);
+                }
+                if ints.dma_outfifo_empty_err().bit() {
+                    res.insert(SpiInterrupt::DmaOutfifoEmptyErr);
+                }
+                if ints.dma_seg_trans_done().bit() {
+                    res.insert(SpiInterrupt::DmaSegTransDone);
+                }
+                if ints.seg_magic_err().bit() {
+                    res.insert(SpiInterrupt::SegMagicErr);
+                }
+                if ints.mst_rx_afifo_wfull_err().bit() {
+                    res.insert(SpiInterrupt::RxAfifoWfullErr);
+                }
+                if ints.mst_tx_afifo_rempty_err().bit() {
+                    res.insert(SpiInterrupt::TxAfifoRemptyErr);
+                }
+                if ints.app2().bit() {
+                    res.insert(SpiInterrupt::App2);
+                }
+                if ints.app1().bit() {
+                    res.insert(SpiInterrupt::App1);
+                }
+            } else if #[cfg(esp32s2)] {
+                if reg_block.slave().read().trans_done().bit() {
+                    res.insert(SpiInterrupt::TransDone);
+                }
+                if reg_block.hold().read().dma_seg_trans_done().bit() {
+                    res.insert(SpiInterrupt::DmaSegTransDone);
+                }
+                if reg_block.slv_rdbuf_dlen().read().seg_magic_err().bit() {
+                    res.insert(SpiInterrupt::SegMagicErr);
+                }
+            } else {
+                if reg_block.slave().read().trans_done().bit() {
+                    res.insert(SpiInterrupt::TransDone);
+                }
+            }
         }
 
         res
     }
 
     /// Resets asserted interrupts
-    #[cfg(gdma)]
     fn clear_interrupts(&self, interrupts: EnumSet<SpiInterrupt>) {
         let reg_block = self.register_block();
-
-        reg_block.dma_int_clr().write(|w| {
-            for interrupt in interrupts {
-                match interrupt {
-                    SpiInterrupt::TransDone => w.trans_done().clear_bit_by_one(),
-                };
+        cfg_if::cfg_if! {
+            if #[cfg(gdma)] {
+                reg_block.dma_int_clr().write(|w| {
+                    for interrupt in interrupts {
+                        match interrupt {
+                            SpiInterrupt::TransDone => w.trans_done().clear_bit_by_one(),
+                            SpiInterrupt::DmaInfifoFullErr => w.dma_infifo_full_err().clear_bit_by_one(),
+                            SpiInterrupt::DmaOutfifoEmptyErr => {
+                                w.dma_outfifo_empty_err().clear_bit_by_one()
+                            }
```

Comment:
Arguably the `*Err` interrupts should be exposed as errors to the user at the end of the transfer, similar to how `I2c` does it (i.e. time out, ack error, arbitration lost, etc).
Allowing the user to clear them here before the transfer is done would get in the way of that.
It's not immediately clear (no pun intended) to me what the best way forward is.
A possibility is to hide them but would a user be interested in knowing about these errors before the transfer is done? I can't think of a use case but it's not a reason to hide 'em.

Reply:
I agree - I cannot think of a use case for handling error-interrupts. I guess we can just remove them from the enum